docs/java-rest/high-level/ml/preview-datafeed.asciidoc

```diff
@@ -0,0 +1,53 @@
+[[java-rest-high-x-pack-ml-preview-datafeed]]
+=== Preview Datafeed API
+
+The Preview Datafeed API provides the ability to preview a {ml} datafeed's data
+in the cluster. It accepts a `PreviewDatafeedRequest` object and responds
+with a `PreviewDatafeedResponse` object.
+
+[[java-rest-high-x-pack-ml-preview-datafeed-request]]
+==== Preview Datafeed Request
+
+A `PreviewDatafeedRequest` object is created referencing a non-null `datafeedId`.
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/MlClientDocumentationIT.java[x-pack-ml-preview-datafeed-request]
+--------------------------------------------------
+<1> Constructing a new request referencing an existing `datafeedId`
+
+[[java-rest-high-x-pack-ml-preview-datafeed-execution]]
+==== Execution
+
+The request can be executed through the `MachineLearningClient` contained
+in the `RestHighLevelClient` object, accessed via the `machineLearningClient()` method.
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/MlClientDocumentationIT.java[x-pack-ml-preview-datafeed-execute]
+--------------------------------------------------
+<1> The raw BytesReference of the data preview
+<2> A List of Map<String,Object> that represents the previewed data
```

Comment:
Similarly `List` and `Map<String, Object>` would look better in backticks.

Also, I think we usually put a space after the comma in `Map<String, Object>`.  It's not so important for private code but it's nice to make the public docs completely consistent.